web/ui/react-app/src/components/SanitizeHTML/SanitizeHTML.test.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import SanitizeHTML from '.';
+
+describe('SanitizeHTML', () => {
+  it(`renders allowed html`, () => {
+    const props = {
+      allowedTags: ['strong'],
+    };
+    const html = shallow(
+      <SanitizeHTML {...props}>{'<strong>text</strong>'}</SanitizeHTML>,
+    );
+    const elem = html.find('div');
+    expect(elem).toHaveLength(1);
+    expect(elem.html()).toEqual(`<div><strong>text</strong></div>`);
+  });
+
+  it('does not render disallowed tags', () => {
+    const props = {
+      tag: 'span' as keyof JSX.IntrinsicElements,
+      allowedTags: ['strong'],
+    };
+    const html = shallow(
+      <SanitizeHTML {...props}>
+        {'<a href="www.example.com">link</a>'}
+      </SanitizeHTML>,
+    );
+    const elem = html.find('span');
+    expect(elem.html()).toEqual('<span>link</span>');
```

Comment:
Oh, good that you're testing this. This isn't actually the behavior we want. We should just escape HTML like our existing `escapeHTML` utility function is doing, not strip away disallowed tags completely.

For example, if someone enters the query `foo{bar="<b>bold</b>"}`, it will now show up as `foo{bar="bold"}` in the query history instead of as the original text.

/cc @boyskila

Reply:
Yes, sanitizeHTML will remove disallowed tags completely. escapeHTML function in Graph.tsx will convert ```foo{bar="<b>bold</b>"}```  to ```foo{bar=&quot;&lt;b&gt;bold&lt;&#x2F;b&gt;&quot;}```
@juliusv  Is this the desired behaviour?

Reply:
@boyskila Yep! So to be clear, we want to HTML-escape it like you said, but in the dropdown the escaped content should be shown as `foo{bar="<b>bold</b>"}` of course.

Reply:
I'm still not sure I what the `SanitizeHTML` should do. Do we want it to do this (not allowing any html)?
```
const SanitizeHTML = ({ tag: Tag = 'div', children, ...rest }: SanitizeHTMLProps) => (
  <Tag {...rest} dangerouslySetInnerHTML={{ __html: escapeHTM(children) }} />
);
```

Reply:
Hmm, so the challenge is actually that the portions of autosuggest items that match the current input are marked in bold / strong in the dropdown: https://github.com/prometheus/prometheus/blob/3b39f6ae4594188bc5ded02a0646de8d1479ccf2/web/ui/react-app/src/ExpressionInput.tsx#L79-L80

...and this particular HTML should be actually shown as is, but any other HTML that was part of a saved (past) query itself should be escaped. Maybe we need to HTML-escape the search items before we inject the partial-match-indicating `<strong>` tags. But that might break the `fuzzy.filter()` search (if a user types `<b>` it might not match the HTML-escaped version of `foo{bar="<b>bold</b>"}`.

Maybe we should defer this question to another PR to ensure we get it right?

Reply:
Gotcha. Yes, let's defer to a separate PR.